### ISSUE TITLE
Please don't put scarf seam settings in the filament profile.

### ISSUE THREAD

**Issue by Cenwulf** [bambulab/BambuStudio#5287]
Please revert the scarf seam settings location back to the process settings as they were prior to the 1.10 release.

Filament settings are, in generally, a set-and-forget affair. The vast majority of tweaking on per print basis should be done in the process settings. Scarf seam settings still are, and arguably always will be, more dependent on the model being printed than the filament being used. In particular having the scarf type tucked away in filament settings is completely nonsensical. Whether a user wants a scarf seam on a contour, a hole, or both has very little to do with the filament.

It is possible that in the future scarf seam settings will get to the point where they can be a set-and-forget type of setting, and can handle any type of model or situation thrown at them. However, they're absolutely no where near that stage yet as evidenced by the fact they've been disabled by default in the latest release. Even if they were to reach this stage of development, there is a strong argument to be made that they still do not belong in the filament settings.

**Comment by MartinChan3** [comment]
I believe the developer enabled the Scarf seam feature by default in V1.10.0 and temporarily disabled it mainly because they are seeking a better solution to address the inconsistency in surface performance, as described in the release notes. Based on that, the Scarf seam will be associated with filaments rather than plates or specific models. Therefore, they have moved the scarf parameters to the filament setting, hoping that it could become a set-and-forget type of feature, which I think is reasonable. We should give them some time to see if they can achieve this goal before reverting to the previous state.

**Comment by Cenwulf** [comment]
As you point and the patch notes acknowledge,this is still a work in progress feature, which is all the more reason to put those settings back into process settings, at least for now. These sort of changes belong in a beta branch rather than prematurely changing the way users interact with a previously stable and usable feature for a loss of functionality and ease of use.

I understand the developer’s reasoning, I just think it’s fundamentally flawed. The seam settings are far more applicable to the model and so it’s far more appropriate that they remain in the process settings as the classic seam has for its entire existence, which itself was pretty well optimised optimised at this point and still remained a process setting despite that.

Even if the aim to make it set-and-forget is achieved, why does that make it more appropriate for them to reside in the filament profile? Is the filament profile a place for unrelated settings you don’t think the user should need to interact with? (I would say no). What’s wrong with leaving them in process settings and enabling them by default in the existing built-in presets?

By all means have overrides in the filament setting for those niche filaments where scarf seam actually isn’t appropriate (out of interest, do we know of any?). This is a very well established way of doing things, the current route is a departure from that norm and for very little benefit aside from potentially reducing clutter in the process settings (which we already have the Advanced switch is for).

To use the wall generator as an example, I’ve found very little reason to ever use anything other than Arachne. However, the classic wall generator is still the default. Should that be changed to Arachne and placed in the filament profile too? Answer: No. It has about as much to do with the filament as the scarf seam does.

**Comment by Cenwulf** [comment]
Another reason not to put scarf seam settings in filament settings, these settings are unavailable for use with modifier parts. There are instances when printing multipart multicolour prints that scarf seam can cause adhesion issues and bleeding between colours on the first lay, for example, this print where a coin is printed face down in two halves with a 0.2 mm nozzle:
![image](https://github.com/user-attachments/assets/5dd3adfb-4402-40c0-bfff-dac1a0da48a2)
![image](https://github.com/user-attachments/assets/c57f4776-a893-4975-ae75-ceb309119c3c)

In the past, I would add a modifier to to ensure that scarf seam is only applied to the outer edge, now that is not possible and print quality of the inner graphic has suffered as a result.

I cannot hammer home enough that there will never be a situation where putting the scarf seam settings in the filament profile is appropriate for all use cases, and the downsides of keeping it in the process settings do not outweigh that.

**Comment by donhohler** [comment]
Scarf settings should NOT be in the filament settings.

**Comment by liftbag** [comment]
I agree. Scarf seam should be assigned to modifiers. In the filament section it makes no sense.

**Comment by xturgorex** [comment]
Please move the scarf seam settings from the filaments section. Even if scarf seams are developed to a point that they would be enabled by default they still do not belong in the filament section.

**Comment by QingZhangBambu** [comment]
i think adding a control switch to the process may meet your needs.
![image](https://github.com/user-attachments/assets/37b6b8b8-03c3-47fb-af48-5cdaf13f0468)

**Comment by Cenwulf** [comment]
Thank you for the suggestion but I do not believe this is an adequate solution.

Scarf seam type already adequately provided this functionality but now it is in the filament settings. Again, why would the choice between contour, or contour and hole be specific to the type of filament being used?

I'm interested in the hearing the justification for putting these specific settings in the filament profile:
![image](https://github.com/user-attachments/assets/b96efac6-f46c-4589-96db-9ba169996c3b)

**Scarf seam type** - as discussed, nothing to do with the filament, perfectly adequate for enabling and disabling scarf seam behaviour as desired by the user, no need to an additional On/Off setting.
**Scarf start height** - when optimising this setting, logically it would have far more to do with layer height (and indirectly nozzle diameter) and so belongs in the process settings.
**Scarf slope gap** - the tool tip talks about optimising the look of closed loop, the existence of which will depend entirely on the 3d geometry being slice, sounds awfully like something that should reside in the process settings.
**Scarf length** - again something that is most likely going to depend on the 3d geometry being sliced, very little to do with the filament being used.

**Comment by QingZhangBambu** [comment]
@Cenwulf, different filaments may have various results for the same parameters. So it does relate to filaments. But also may get different effects in different parts. You need a method to use a modifier to control the effect for different model parts.  I will figure out that.

**Comment by QingZhangBambu** [comment]
@Cenwulf  And, the scarf should not apply to the first layer, it's a bug, could you provide the 3mf file you used on the instance?

**Comment by Cenwulf** [comment]
@QingZhangBambu
![image](https://github.com/user-attachments/assets/e064f630-099e-4d90-b5d2-968b659d07e0)
I've made some changes to the project since but it is still attempting to do the scarf seam on the first layer.
[trolley_token_keroppi.zip](https://github.com/user-attachments/files/18041790/trolley_token_keroppi.zip)

**Comment by Cenwulf** [comment]
@QingZhangBambu could the scarf seam settings go in the filamanet settings overrides to give the best of both worlds?
![image](https://github.com/user-attachments/assets/a02169f7-7a75-481b-b690-06a546efcea8)

**Comment by QingZhangBambu** [comment]
@Cenwulf I tend to add some params on process preset which can override the filament params.

**Comment by QingZhangBambu** [comment]
I've put this feature to our product managers, they will review it.

**Comment by JohnSmithCYM** [comment]
As @QingZhangBambu said, we will add params on process preset which can override params under filament settings.

**Comment by Cenwulf** [comment]
Thanks for the updates. Having process setting that override settings in filament profile seems opposite to the established norm? Is there previdente for this?

Regardless, as long is the end result is that the user is able to fully control the application of scarf seams using the process setting alone, that’s all that really matters. 

Thanks again.

**Comment by tjc2005** [comment]
are they going to revert to the old process or not? this is so stupid having it in the filament settings.

**Comment by bcorcoran** [comment]
Please remove scarf settings from filament profiles. @QingZhangBambu please provide an update!

**Comment by QingZhangBambu** [comment]
I have fixed the issue. The product manager will decide which version to release. I will confirm the timing with him.

**Comment by bcorcoran** [comment]
> I have fixed the issue. The product manager will decide which version to release. I will confirm the timing with him.

2.0.0 is out and does not include this fix. Is it going to be in a future release?

**Comment by tjc2005** [comment]
> I have fixed the issue. The product manager will decide which version to release. I will confirm the timing with him.

This is not fixed

**Comment by BrainStone** [comment]
While I can understand why you would tie certain scarf seam settings to the filament, no longer having a per project setting is plainly stupid.

The use case to decide against or for it based on the use of the model (functional prints vs aesthetic prints) is much more prevalent than tying it to a filament. Especially considering it defaults to being off. Which means I now need to switch 2 profiles if I want to switch between my quality and speed setting. For basically no benefit.

**Comment by bcorcoran** [comment]
> While I can understand why you would tie certain scarf seam settings to the filament, no longer having a per project setting is plainly stupid.
> 
> The use case to decide against or for it based on the use of the model (functional prints vs aesthetic prints) is much more prevalent than tying it to a filament. Especially considering it defaults to being off. Which means I now need to switch 2 profiles if I want to switch between my quality and speed setting. For basically no benefit.
> 
> This is especially infuriating because now I need to have a separate filament profile which adds several clicks to selecting the filament from the AMS and very easily causes the setting to be just off when it should be on.

I actually do not understand why you would ever want this as a filament-specific setting. It's entirely project/model dependent, so the filament is irrelevant. If you are using a filament that doesn't work well with scarf seams, you will probably already know that.

I switched to Orca Slicer and LAN mode until they revert this change. Yes, specifically for this feature.
